pkg/command/eventing/driver/redis_sender.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2020 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package driver
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/cloudevents/sdk-go/v2/types"
+
+	redisParse "github.com/go-redis/redis/v8"
+	"github.com/gomodule/redigo/redis"
+)
+
+type RedisEventSender struct {
+	Plan    SendEventsPlan
+	Address string
+	// NumConsumers   string
+	TLSCertificate string
+	Stream         string
+	// Group          string
+	// PodName        string
+	pool *redis.Pool
+}
+
+func NewRedisSender(plan SendEventsPlan) EventSender {
+	rs := &RedisEventSender{
+		Plan:           plan,
+		Address:        "rediss://localhost:6379",
+		TLSCertificate: "",
+		Stream:         "mystream",
+	}
+	rs.pool = newPool(rs.Address, rs.TLSCertificate)
+	// rs.logger :=logging.FromContext(ctx).Desugar().With(zap.String("stream", config.Stream)),
+	//logger: logger
+	return rs
+}
+
+func strptr(s string) *string {
+	return &s
+}
+
+func (s RedisEventSender) Send() EventsStats {
+	plan := s.Plan
+	senderName := plan.senderName
+	//values := map[string]string{"id": "1234668888888", "source": "323223232332909090", "type": "dev.knative.eventing.test.scaling", "timestamp": "12929299999992222"}
+	source := types.URIRef{URL: url.URL{Scheme: "http", Host: "example.com", Path: "/source"}}
+	timestamp := types.Timestamp{Time: time.Now()}
+	//schema := types.URI{URL: url.URL{Scheme: "http", Host: "example.com", Path: "/schema"}}
+	e := event.Event{
+		Context: event.EventContextV1{
+			Type:   "com.example.FullEvent",
+			Source: source,
+			ID:     "full-event",
+			Time:   &timestamp,
+			//DataSchema: &schema,
+			Subject: strptr("topic"),
+		}.AsV1(),
+	}
+	//if err := event.SetData("text/json", "[\"fruit\", \"orange\"]"); err != nil {
+	data := []byte("[\"fruit\", \"orange\"]")
+	//data := []byte("{\"a\" : \"b\"}") // {"a": "b"}
+	if err := e.SetData(event.ApplicationJSON, data); err != nil {
+		panic(err)
```

Comment:
Same as before these panic and logFatal will come back to hunt you when you need to debug. Best just deal with error at first catch with message, return and cascade up to end cleanly